server/server.go

```diff
@@ -35,9 +95,94 @@ func servePages(e *echo.Echo) {
 }
 
 func serveAPI(e *echo.Echo) {
+	//Set client options
+	clientOptions := options.Client().ApplyURI("mongodb://localhost:27017")
+	//Connect to MongoDB
+	client, err := mongo.Connect(context.TODO(), clientOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+	//Check connection
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Creating collections
+	postsCollection := client.Database("csesoc").Collection("posts")
+	catCollection := client.Database("csesoc").Collection("categories")
+	sponsorCollection := client.Database("csesoc").Collection("sponsors")
+	userCollection := client.Database("csesoc").Collection("users")
+
 	// Add more API routes here
 	e.GET("/api/v1/test", func(c echo.Context) error {
 		return c.String(http.StatusOK, "Hello, World!")
 	})
 
+	e.POST("/login/", login(userCollection))
+
+	// Routes for posts
+	e.GET("/post/:id/", getPost(postsCollection))
+	e.GET("/posts/", getAllPosts(postsCollection))
+	e.POST("/post/", newPost(postsCollection))
+	e.PUT("/post/:id/", updatePost(postsCollection))
+	e.DELETE("/post/:id/", deletePost(postsCollection))
+
+	// Routes for categories
+	e.GET("/category/:id/", getCat(catCollection))
+	e.POST("/category/", newCat(catCollection))
+	e.PATCH("/category/", patchCat(catCollection))
+	e.DELETE("/category/", deleteCat(catCollection))
+
+	// Routes for sponsors
+	e.POST("/sponsor/", newSponsor(sponsorCollection))
+	e.DELETE("/sponsor/", deleteSponsor(sponsorCollection))
+}
+
+func login(collection *mongo.Collection) echo.HandlerFunc {
+	return auth(c echo.Context)
+}
```

Comment:
These functions should have a structure more like (recall these functions return `echo.HandlerFunc`):
```go
func login(collection *mongo.Collection) echo.HandlerFunc {
return func(c echo.Context) error {
        // It may be a wise idea to return a status code from your function so you can verify the action actually succeeded before returning a 200 OK.
        tokenString = NewPost(collection);
        return c.JSON(http.StatusOK, H{
		"token": tokenString,
	})

}
}
```